projections/shapes/lib/shapes/schema.rb

```diff
@@ -57,28 +53,29 @@ module Schema
     OperationInputOutput.add_member(:structure, 'structure', Structure)
     OperationInputOutput.add_member(:union, 'union', Union)
     OperationInputOutput.type = Types::OperationInputOutput
-
-    Structure.add_member(:member, 'member', String, traits: {"smithy.ruby.tests#shape" => {}})
+    Structure.add_member(:member, 'member', String, traits: {"smithy.ruby.tests#shape"=>{}})
     Structure.type = Types::Structure
-
-    Union.add_member(:string, 'string', String, Types::Union::String, traits: {"smithy.ruby.tests#shape" => {}})
-    Union.add_member(:structure, 'structure', Structure, Types::Union::Structure, traits: {"smithy.ruby.tests#shape" => {}})
-    Union.add_member(:unit, 'unit', Prelude::Unit, Types::Union::Unit, traits: {"smithy.ruby.tests#shape" => {}})
+    Union.add_member(:string, 'string', String, Types::Union::String, traits: {"smithy.ruby.tests#shape"=>{}})
+    Union.add_member(:structure, 'structure', Structure, Types::Union::Structure, traits: {"smithy.ruby.tests#shape"=>{}})
+    Union.add_member(:unit, 'unit', Prelude::Unit, Types::Union::Unit, traits: {"smithy.ruby.tests#shape"=>{}})
     Union.add_member(:unknown, 'unknown', Prelude::Unit, Types::Union::Unknown)
     Union.type = Types::Union
 
     SERVICE = ServiceShape.new do |service|
       service.id = "smithy.ruby.tests#ShapeService"
       service.name = "ShapeService"
       service.version = "2018-10-31"
-      service.traits = {"smithy.ruby.tests#shape" => {}}
+      service.traits = {"smithy.ruby.tests#shape"=>{}}
       service.add_operation(:operation, OperationShape.new do |operation|
         operation.id = "smithy.ruby.tests#Operation"
         operation.name = "Operation"
         operation.input = OperationInputOutput
         operation.output = OperationInputOutput
-        operation.traits = {"smithy.ruby.tests#shape" => {}}
+        operation.traits = {"smithy.ruby.tests#shape"=>{}}
       end)
     end
+
+    TYPE_REGISTRY = Smithy::Schema::TypeRegistry.new
+    TYPE_REGISTRY.register(OperationInputOutput, Structure, Union)
```

Comment:
I THINK we only care about shapes that has a type representation so I only include those specific ones.